packages/core/src/charts/meter.ts

```diff
@@ -49,39 +49,61 @@ export class MeterChart extends Chart {
 	}
 
 	getComponents() {
-		// Specify what to render inside the graph only
-		const graph = {
-			id: 'meter-graph',
-			components: [new Meter(this.model, this.services)],
-			growth: LayoutGrowth.STRETCH,
-			renderType: RenderTypes.SVG,
-		};
+		const options = this.model.getOptions();
+		const meterComponents = [
+			// Meter has a unique dataset title within the graph
+			{
+				id: 'meter-title',
+				components: [new MeterTitle(this.model, this.services)],
+				growth: LayoutGrowth.STRETCH,
+				renderType: RenderTypes.SVG,
+			},
+			// Create the title spacer
+			{
+				id: 'spacer',
+				components: [
+					new Spacer(this.model, this.services, { size: 8 }),
+				],
+				growth: LayoutGrowth.STRETCH,
+			},
+			// Specify what to render inside the graph only
+			{
+				id: 'meter-graph',
+				components: [new Meter(this.model, this.services)],
+				growth: LayoutGrowth.STRETCH,
+				renderType: RenderTypes.SVG,
+			},
+		];
+
+		//breakdownFormatter
+		const breakdownFormatter = Tools.getProperty(
+			options,
+			'meter',
+			'proportional',
+			'breakdownFormatter'
+		);
 
-		// Meter has an unique dataset title within the graph
-		const titleComponent = {
-			id: 'meter-title',
-			components: [new MeterTitle(this.model, this.services)],
-			growth: LayoutGrowth.STRETCH,
-			renderType: RenderTypes.SVG,
-		};
+		// totalFormatter function
+		const totalFormatter = Tools.getProperty(
+			options,
+			'meter',
+			'proportional',
+			'totalFormatter'
+		);
 
-		// create the title spacer
-		const titleSpacerComponent = {
-			id: 'spacer',
-			components: [new Spacer(this.model, this.services, { size: 8 })],
-			growth: LayoutGrowth.STRETCH,
-		};
+		// Check to see if the formatter functions exist
+		if (!!breakdownFormatter && !!totalFormatter) {
+			// Check if both formatters return null or empty string, do not render them
+			if (!breakdownFormatter(null) && !totalFormatter(null)) {
```

Comment:
should we pass `null` in here? sounds like that could cause errors on the user's side, e.g. if they assume they'll get a string and try to do something like `str.replace()`...

Reply:
You're right, for that reason I have decided that it would make more sense to simply add an option to proportional meter chart. This would do two things:

1. Would not cause a breaking change for users that are returning null value from total and breakdown formatters (Previously, it would require them to perhaps make css adjustments since the padding would be gone)
2. Less computation & no chance of errors like previously

```js
proportional?: {
    ...
    renderTitle?: boolean;
};
```

Reply:
does `renderTitle` just remove the chart title? If so, we should add it to all the charts, correct?

Reply:
e.g. `{ title: { visible: false } }` or something like that

Reply:
> does `renderTitle` just remove the chart title? If so, we should add it to all the charts, correct?

Yes, it removes the Meter Title (Total and breakdown) specifically not the chart title.

> does `renderTitle` just remove the chart title? If so, we should add it to all the charts, correct?
> e.g. `{ title: { visible: false } }` or something like that

Currently, there is a condition in code that renders the title only if it exists in `chart.ts`:

```ts
		if (titleAvailable || toolbarEnabled) {

// Line below adds the header (title) to the layout only if the title value exists. If you don't pass in a truthy title, 
// it will not render
			topLevelLayoutComponents.push(headerComponent);

			const titleSpacerComponent = {
				id: 'spacer',
				components: [
					new Spacer(
						this.model,
						this.services,
						toolbarEnabled ? { size: 15 } : undefined
					),
				],
				growth: LayoutGrowth.PREFERRED,
			};

			topLevelLayoutComponents.push(titleSpacerComponent);
		}

```